src/opencl_common.c

```diff
@@ -88,6 +88,54 @@ extern int autotune_get_next_gws_size(size_t num, int step, int startup,
                                       int default_value);
 extern int autotune_get_prev_gws_size(size_t num, int step);
 
+/*
+ * List entry, allocated as (sizeof(struct kernel_opencl) + sizeof(data)).
+ */
```

Comment:
To avoid duplication, I can use pseudo-generics. But, the source code becomes hard to follow.